Fix TestWorld.evolveMigration

DavidKerkmann opened the pull request:
# Changes and Information

Please **briefly list the changes** made, additional Information and what the Reviewer should look out for:

- Fixed the TestWorld.evolveMigration unit test, that was most likely an issue created by #624 where new tests were added.
- Removed the parameters, since they were actually not providing much and were on the wrong AgeGroup partially.

## Merge Request - Guideline Checklist

Please check our [git workflow](https://github.com/DLR-SC/memilio/wiki/git-workflow). Use the **draft** feature if the Pull Request is not yet ready to review.

### Checks by code author

- [x] Every addressed issue is linked (use the "Closes #ISSUE" keyword below)
- [x] New code adheres to [coding guidelines](https://github.com/DLR-SC/memilio/wiki/Coding-guidelines)
- [x] No large data files have been added (files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)
- [x] Tests are added for new functionality and a local test run was successful
- [x] Appropriate **documentation** for new functionality has been added (Doxygen in the code and Markdown files if necessary)
- [x] Proper attention to licenses, especially no new third-party software with conflicting license has been added

### Checks by code reviewer(s)

- [ ] Corresponding issue(s) is/are linked and addressed
- [ ] Code is clean of development artifacts (no deactivated or commented code lines, no debugging printouts, etc.)
- [ ] Appropriate **unit tests** have been added, CI passes and code coverage is acceptable (did not decrease)
- [ ] No large data files added in the whole history of commits(files should in sum not exceed 100 KB, avoid PDFs, Word docs, etc.)

DavidKerkmann commented:
I put p1 back to InfectedNoSymptoms to verify that this person still goes to work without symptoms.

dabele commented on cpp/tests/test_abm_world.cpp:
```diff
@@ -242,8 +233,22 @@ TEST(TestWorld, evolveMigration)
 
-        auto& p1 =
-            add_test_person(world, home_id, mio::abm::AgeGroup::Age15to34, mio::abm::InfectionState::Susceptible, t);
+        ScopedMockDistribution<testing::StrictMock<MockDistribution<mio::UniformDistribution<double>>>>
+            mock_uniform_dist;
+        EXPECT_CALL(mock_uniform_dist.get_mock(), invoke)
+            .Times(testing::AtLeast(8))
+            .WillOnce(testing::Return(0.8)) // draw random work group
+            .WillOnce(testing::Return(0.8)) // draw random school group
+            .WillOnce(testing::Return(0.8)) // draw random work hour
+            .WillOnce(testing::Return(0.8)) // draw random school hour
+            .WillOnce(testing::Return(0.8)) // draw random work group
+            .WillOnce(testing::Return(0.8)) // draw random school group
+            .WillOnce(testing::Return(0.8)) // draw random work hour
+            .WillOnce(testing::Return(0.8)) // draw random school hour
+            .WillRepeatedly(testing::Return(1.0)); // this forces p1 and p3 to recover
+
+        auto& p1 = add_test_person(world, home_id, mio::abm::AgeGroup::Age15to34,
+                                   mio::abm::InfectionState::InfectedNoSymptoms, t);
```
I think the fact that we need to setup a mock during the setup of the test (not the actual thing to be tested) points to problems in the design. I think it would be better if there is always a way to set a specific state. This makes not just testing easier/clearer but also running very specific scenarios. Of course the ability to randomize things should exist as well, maybe always have two overloads, one with RNG, one with fixed values.

Some instances where this is a problem (maybe others):
- infections
- set up of persons (random work group)

These cases are more obvious now that you always need to pass in an RNG as an argument. Nothing to be done now, would be too big, but maybe we should take this into account in future development.

DavidKerkmann replied:
I guess you are right, we should add that at some point soon.

dabele commented:
Sorry, I also removed the `std::cout` stuff in my Pull Request that just got merged, looks like you need to resolve conflicts. I would have merged this first and resolved them myself, but Martin was faster :)

DavidKerkmann commented:
> Sorry, I also removed the `std::cout` stuff in my Pull Request that just got merged, looks like you need to resolve conflicts. I would have merged this first and resolved them myself, but Martin was faster :)

No problem, I have changed this. Somehow, the formatting on some lines is different on my autosave. Not sure why that is.
